arches/app/datatypes/datatypes.py

```diff
@@ -1911,71 +1912,18 @@ def validate(self, value, row_number=None, source="", node=None, nodeid=None, st
-                    else related_resource["resourceXresourceId"]
-                )
-                defaults = {
-                    "resourceinstanceidfrom_id": tile.resourceinstance_id,
-                    "resourceinstanceidto_id": related_resource["resourceId"],
-                    "notes": "",
-                    "relationshiptype": related_resource["ontologyProperty"],
-                    "inverserelationshiptype": related_resource["inverseOntologyProperty"],
-                    "tileid_id": tile.pk,
-                    "nodeid_id": nodeid,
-                }
-                if related_resource["ontologyProperty"] == "" or related_resource["inverseOntologyProperty"] == "":
-                    if models.ResourceInstance.objects.filter(pk=related_resource["resourceId"]).exists():
-                        target_graphid = str(models.ResourceInstance.objects.get(pk=related_resource["resourceId"]).graph_id)
-                        for graph in models.Node.objects.get(pk=nodeid).config["graphs"]:
-                            if graph["graphid"] == target_graphid:
-                                if related_resource["ontologyProperty"] == "":
-                                    try:
-                                        defaults["relationshiptype"] = graph["ontologyProperty"]
-                                    except:
-                                        pass
-                                if related_resource["inverseOntologyProperty"] == "":
-                                    try:
-                                        defaults["inverserelationshiptype"] = graph["inverseOntologyProperty"]
-                                    except:
-                                        pass
-                try:
-                    rr = models.ResourceXResource.objects.get(pk=resourceXresourceId)
-                    for key, value in defaults.items():
-                        setattr(rr, key, value)
-                    rr.save()
-                except models.ResourceXResource.DoesNotExist:
-                    rr = models.ResourceXResource(**defaults)
-                    rr.save()
-                related_resource["resourceXresourceId"] = str(rr.pk)
-                resourceXresourceSaved.add(rr.pk)
-
-            # get a list of all resourceXresources with the same tile and node
-            # if there are any ids in that list that aren't in the resourceXresourceSaved
-            # then those need to be removed from the db
-            resourceXresourceInDb = set(
-                models.ResourceXResource.objects.filter(tileid_id=tile.pk, nodeid_id=nodeid).values_list("pk", flat=True)
-            )
-            to_delete = resourceXresourceInDb - resourceXresourceSaved
-            for rr in models.ResourceXResource.objects.filter(pk__in=to_delete):
-                rr.delete()
+        with connection.cursor() as cursor:
+            cursor.execute(sql)
+            ret = cursor.fetchone()
```

Comment:
In order for this to work from the web UI the resource relationship would need to be indexed. Either that or we would need to stop relying on relationship index and get our relationships from the database.

Reply:
Good catch @chiatt!  I remembered that we indexed that data when we saved a ResourceXResource instance but of course that was when we were using the ORM.  
Honestly I'm not sure why we're indexing that data at all.  The indexed data has the same form as the table itself.  This is where [pgsync](https://pgsync.com/) might pay off.  
In the mean time I should probably fix that....

Reply:
I think ES was used originally to minimize database queries because our db query count was really, really high at the time. That isn't the case anymore, so I think getting relationship data from the database and removing the resource relationship index altogether would be a better approach than indexing with pgsync. That is unless we have a reason to search within resource relationships in a way that we can't do with postgres. I can't think of what that query would be though.